lib/src/iterable_extensions.dart

```diff
@@ -549,6 +550,54 @@ extension IterableExtension<T> on Iterable<T> {
     }
     return true;
   }
+
+  /// Returns an iterable whose elements are contiguous slices of [this].
+  ///
+  /// Each slice is [length] elements long, except for the last one which may be
+  /// shorter if [this] contains too few elements. Each slice begins after the
+  /// last one ends.
+  ///
+  /// For example, `(1, 2, 3, 4, 5).slices(2)` returns `((1, 2), (3, 4), (5))`.
+  Iterable<Iterable<T>> slices(int length) sync* {
+    if (length < 1) throw RangeError.range(length, 1, null, 'length');
+
+    var iterator = this.iterator;
+    while (iterator.moveNext()) {
+      yield () sync* {
+        var i = 0;
+        do {
+          yield iterator.current;
+          i++;
+        } while (i < length && iterator.moveNext());
+      }();
+    }
+  }
+
+  /// Returns an iterable of each subsequence of [this] of the given [length].
+  ///
+  /// Each subsequence is exactly [length] elements long and shares `length - 1`
+  /// elements with the previous subsequence. Returns an empty iterable if
+  /// [length] is greater than this iterable's length.
+  ///
+  /// For example, `(1, 2, 3, 4).subsequences(2)` returns
+  /// `([1, 2], [2, 3], [3, 4])`.
+  Iterable<List<T>> subsequences(int length) sync* {
```

Comment:
The name also is not particularly descriptive. Maybe `allSubsequences`?

Reply:
I'm not sure "all" adds much descriptive power there... if you wanted it to be more explicit, it could be `subsequencesOfLength`, but that's a bit redundant with the parameter name.